packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -241,28 +367,50 @@ def imshow(
             layout["xaxis"] = dict(scaleanchor="y", constrain="domain")
             layout["yaxis"]["constrain"] = "domain"
         colorscale_validator = ColorscaleValidator("colorscale", "imshow")
-        if zmin is not None and zmax is None:
-            zmax = img.max()
-        if zmax is not None and zmin is None:
-            zmin = img.min()
-        range_color = range_color or [zmin, zmax]
         layout["coloraxis1"] = dict(
             colorscale=colorscale_validator.validate_coerce(
                 args["color_continuous_scale"]
             ),
             cmid=color_continuous_midpoint,
-            cmin=range_color[0],
-            cmax=range_color[1],
+            cmin=zmin,
+            cmax=zmax,
         )
         if labels["color"]:
             layout["coloraxis1"]["colorbar"] = dict(title_text=labels["color"])
 
     # For 2D+RGB data, use Image trace
-    elif img.ndim == 3 and img.shape[-1] in [3, 4]:
-        if zmax is None and img.dtype is not np.uint8:
-            zmax = _infer_zmax_from_type(img)
-        zmin, zmax = _vectorize_zvalue(zmin), _vectorize_zvalue(zmax)
-        trace = go.Image(z=img, zmin=zmin, zmax=zmax)
+    elif img.ndim == 3 and img.shape[-1] in [3, 4] or (img.ndim == 2 and binary_string):
+        rescale_image = True  # to check whether image has been modified
+        if zmin is not None and zmax is not None:
+            zmin, zmax = _vectorize_zvalue(zmin), _vectorize_zvalue(zmax)
+        if binary_string:
+            if zmin is None and zmax is None:  # no rescaling, faster
+                img_rescaled = img
+                rescale_image = False
```

Comment:
here we could also set zmax to [255, 255, 255, 255] if no new colormodel is introduced in plotly.js